kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaServiceReconciler.java

```diff
@@ -8,52 +8,200 @@
+                        KafkaService.class,
+                        service -> asRef(service.getSpec().getTls().getTrustAnchorRef())))
+                .build();
+        return List.of(
+                new InformerEventSource<>(serviceToSecret, context),
+                new InformerEventSource<>(serviceToConfigMap, context));
+    }
+
     @Override
     public UpdateControl<KafkaService> reconcile(KafkaService service, Context<KafkaService> context) {
-        UpdateControl<KafkaService> uc = UpdateControl.patchStatus(statusFactory.newTrueConditionStatusPatch(service, Condition.Type.Accepted));
+
+        KafkaService updatedService = null;
+
+        var trustAnchorRefOpt = Optional.ofNullable(service.getSpec())
+                .map(KafkaServiceSpec::getTls)
+                .map(Tls::getTrustAnchorRef);
+        if (trustAnchorRefOpt.isPresent()) {
+            updatedService = checkTrustAnchorRef(service, context, trustAnchorRefOpt.get());
+        }
+
+        if (updatedService == null) {
+            var certRefOpt = Optional.ofNullable(service.getSpec())
+                    .map(KafkaServiceSpec::getTls)
+                    .map(Tls::getCertificateRef);
+            if (certRefOpt.isPresent()) {
+                updatedService = checkCertRef(service, context, certRefOpt.get());
+            }
+        }
+
+        if (updatedService == null) {
+            updatedService = statusFactory.newTrueConditionStatusPatch(service, ResolvedRefs);
+        }
+
+        UpdateControl<KafkaService> uc = UpdateControl.patchStatus(updatedService);
+
         if (LOGGER.isInfoEnabled()) {
             LOGGER.info("Completed reconciliation of {}/{}", namespace(service), name(service));
         }
         return uc;
     }
 
-    @Override
-    public List<EventSource<?, KafkaService>> prepareEventSources(EventSourceContext<KafkaService> context) {
-        return io.javaoperatorsdk.operator.api.reconciler.Reconciler.super.prepareEventSources(context);
+    @Nullable
+    private KafkaService checkTrustAnchorRef(KafkaService service,
+                                             Context<KafkaService> context,
+                                             TrustAnchorRef trustAnchorRef) {
+        String path = "spec.tls.trustAnchorRef";
+        if ("ConfigMap".equals(Optional.ofNullable(trustAnchorRef.getKind()).orElse("ConfigMap"))
+                && Optional.ofNullable(trustAnchorRef.getGroup()).orElse("").isEmpty()) {
+            Optional<ConfigMap> configMapOpt = context.getSecondaryResource(ConfigMap.class, CONFIG_MAPS_EVENT_SOURCE_NAME);
+            if (configMapOpt.isEmpty()) {
+                return statusFactory.newFalseConditionStatusPatch(service, ResolvedRefs,
+                        Condition.REASON_REFS_NOT_FOUND,
+                        path + ": referenced resource not found");
+            }
+            else {
+                String key = trustAnchorRef.getKey();
+                if (key == null) {
```

Comment:
Strictly speaking this should be enforced from the CRD schema, but it seems harmless to have a check in the code.